src/utils/generateAlphabeticName.js

```diff
@@ -1,17 +1,25 @@
 // @flow
-const chars = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('')
-const charsLength = chars.length
+/* eslint-disable no-bitwise */
 
-/* Some high number, usually 9-digit base-10. Map it to base-😎 */
+/* This is the "capacity" of our alphabet i.e. 2x26 for all letters plus their capitalised
+ * counterparts */
+const charsLength = 52
+
+/* start at 75 for 'a' until 'z' (25) and then start at 65 for capitalised letters */
+const getAlphabeticChar = (code: number): string =>
+  String.fromCharCode(code + (code > 25 ? 39 : 97))
+
+/* input a number, usually a hash and convert it to base-52 */
 const generateAlphabeticName = (code: number): string => {
   let name = ''
   let x
 
-  for (x = code; x > charsLength; x = Math.floor(x / charsLength)) {
-    name = chars[x % charsLength] + name
+  /* get a char and divide by alphabet-length (bitwise zero for flooring the number) */
+  for (x = code | 0; x > charsLength; x = (x / charsLength) | 0) {
+    name = getAlphabeticChar(x % charsLength) + name
   }
 
-  return chars[x % charsLength] + name
+  return getAlphabeticChar(x % charsLength) + name
```

Comment:
Is this refactor going to change existing classes? If so, that'd be very annoying for any existing tests :confused:

Reply:
Nope, all tests still pass & and the output is the exact same

Reply:
Btw I’ve done this for gzip :) turns out the small list of the alphabet is really bad for gzip since that’s really impossible to compress.